packages/e2e/artillery/StakePoolSearch.ts

```diff
@@ -0,0 +1,94 @@
+import { AfterResponse, BeforeRequest, WhileTrue } from './artillery';
+import { Cardano, StakePoolQueryOptions, StakePoolSearchResults } from '@cardano-sdk/core';
+
+interface StakePoolSearchVars extends StakePoolSearchResults {
+  args: [StakePoolQueryOptions];
+  moreResults: boolean;
+  queryFirst: boolean;
+  queryWithName: boolean;
+  queryWithStatus: boolean;
+  startedAt: number;
+}
+
+const PAGE_SIZE = 100;
+
+const names: Record<string, boolean> = {};
+let firstUser = '';
+
+const getStatName = (vars: StakePoolSearchVars) => {
+  if (vars.queryFirst) return 'query.full_first';
+  if (vars.queryWithStatus) return vars.queryWithName ? 'query.name_status' : 'query.status';
+  return vars.queryWithName ? 'query.name' : 'query.full';
+};
+
+export const after: AfterResponse<StakePoolSearchVars> = async (_params, _response, ctx, ee, done) => {
+  // Take values to randomize next requests
+  for (const result of ctx.vars.pageResults) if (result.metadata) names[result.metadata.name] = true;
+
+  // Send custom metrics
+  ee.emit('histogram', `${getStatName(ctx.vars)}.time`, Date.now() - ctx.vars.startedAt);
+
+  done();
+};
+
+export const before: BeforeRequest<StakePoolSearchVars> = async (_params, ctx, ee, done) => {
+  let options = ctx.vars.args?.[0];
+
+  ctx.vars.startedAt = Date.now();
+
+  // If not already initialized: prepare the new request starting from the first page
+  if (!options) {
+    options = { filters: {}, pagination: { limit: PAGE_SIZE, startAt: 0 } };
+    ctx.vars.args = [options];
+  }
+
+  // If we have a pageResults, this is not the first call of the session: request next page
+  if (ctx.vars.pageResults) options.pagination!.startAt += PAGE_SIZE;
+  // If this is the first call of the session: randomize query parameters
+  else {
+    const namesList = Object.keys(names);
+
+    // If this is the first virtual user, we need to make a query that will fully prime the API
```

Comment:
```suggestion
    // If this is the first virtual user, we need to make a query that will fully prime the API cache
```

![image](https://user-images.githubusercontent.com/303881/189361037-60b54ff3-c9bb-40a5-9069-b751f6e39b77.png)

Reply:
f104f574b should fix this